### PR TITLE
Laboratory work #1, Maria Nikitenko - 23FPL2

### PR DESCRIPTION
**Pull request by MariaNikitenko19**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -20,6 +20,15 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if type(text) is not str:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:23]: Use `isinstance`

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -34,7 +43,13 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if type(tokens) is not list or None in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:46]: the same

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -64,7 +82,12 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if type(predicted) is not list or type(actual) is not list and not len(predicted) == len(actual):
+        return None
+    diff = 0
+    for i in range(len(actual)):
+        diff += (actual[i] - predicted[i])**2
+    return diff/len(actual)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:90]: spaces around `/`

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -64,7 +82,12 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if type(predicted) is not list or type(actual) is not list and not len(predicted) == len(actual):
+        return None
+    diff = 0
+    for i in range(len(actual)):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:88]: use enumerate

File: lab_1_classify_by_unigrams/start.py [hunk 5]
```diff
@@ -1,22 +1,30 @@
 """
 Language detection starter
 """
-# pylint:disable=too-many-locals, unused-argument, unused-variable
 
+# pylint:disable=too-many-locals, unused-argument, unused-variable
+from lab_1_classify_by_unigrams.main import create_language_profile
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:6]: move before pylint disable

File: seminars/practice_2_strings.py [hunk 6]
```diff
@@ -61,7 +61,6 @@ def multiply_string(input_string: str, how_many: int) -> str:
 # multiply_string('Hi', 1) → 'Hi'
 # multiply_string('Hi', 0) → ''
 
-
 # Task 2:
```
Comment (artyomtugaryov) [inline seminars/practice_2_strings.py:64]: revert

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/start.py (hunk 5 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/start.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -34,7 +43,14 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     In case of corrupt input arguments, None is returned
     """
-
+    if not isinstance(tokens, list) or None in tokens\
+            or not all(isinstance(token,str) for token in tokens):
+        return None
+    frequency_dictionary = {}
+    for token in tokens:
+        if token not in frequency_dictionary:
+            frequency_dictionary[token] = tokens.count(token)/len(tokens)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:52]: It is unoptimize. Can you try to call `.count()` only for unique symbols

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -64,7 +87,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
     In case of corrupt input arguments, None is returned
     """
-
+    if not isinstance(predicted, list) or not isinstance(actual, list) or \
+            not len(predicted) == len(actual):
+        return None
+    diff = 0
+    for m, n in enumerate(actual):
+        diff += (n - predicted[m])**2
+    return float(diff / len(actual))
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:96]: Why do we need to cast it?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -64,7 +88,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if not isinstance(predicted, list) or not isinstance(actual, list) or \
+            not len(predicted) == len(actual):
+        return None
+    diff = 0
+    for m, n in enumerate(actual):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:95]: please use meaningful names

File: seminars/practice_2_strings.py [hunk 10]
```diff
@@ -79,7 +78,6 @@ def front_times(input_string: str, how_many: int) -> str:
 # front_times('', 4) → ''
 # front_times('Abc', 0) → ''
 
-
```
Comment (artyomtugaryov) [inline seminars/practice_2_strings.py:82]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
What about 10?

File: lab_1_classify_by_unigrams/main.py (hunk 9 above)
Comment (MariaNikitenko19) [reply lab_1_classify_by_unigrams/main.py]: done

File: seminars/practice_2_strings.py (hunk 10 above)
Comment (MariaNikitenko19) [reply seminars/practice_2_strings.py]: done

File: seminars/practice_2_strings.py (hunk 6 above)
Comment (MariaNikitenko19) [reply seminars/practice_2_strings.py]: done

**Pull request closed without merging**